bump nodejs dependencies based on npm audit

ohadlevy opened the pull request:
this pr ignores jquery 3 update and compression-webpack-plugin as they are handled in other prs and require potential code chagnge

ohadlevy committed:
bump nodejs dependencies based on npm audit

this pr ignores jquery 3 update and compression-webpack-plugin as they are handled in other prs and require potential code chagnge

ohadlevy commented:
katello test failures seem unrelated.

tbrisker commented on package.json:
```diff
@@ -61,7 +61,7 @@
     "brace": "^0.10.0",
     "datatables.net": "~1.10.12",
-    "datatables.net-bs": "~1.10.12",
+    "datatables.net-bs": "^2.1.1",
```
iirc, "datatables.net-bs" is the bootstrap styling for datatables. Are you sure bumping this version without changing the matching "datatables.net" version works?

ohadlevy replied:
I did confirm it works (nothing seems to fail and the page seemed to render correctly).
I do hope we will migrate away from datatables sooner than later anyway..

ekohl replied:
This specific version is marked as deprecated on NPM. Are you sure this is the correct version? https://www.npmjs.com/package/datatables.net-bs lists 1.10.16 as the current version.

ekohl commented on package.json:
```diff
@@ -39,16 +39,16 @@
     "jsdom": "^9.5.0",
     "lodash-webpack-plugin": "^0.11.4",
-    "node-sass": "^4.5.0",
+    "node-sass": "^4.9.0",
```
This is a rather complicated package to build. Could we submit this as a separate PR so we can get all the other changes in first?

ekohl reviewed (commented):
I've opened https://github.com/theforeman/foreman-packaging/pull/2673. This is the first package we handle where something depends on `@x/y` where everything else just had `x` as names. Let's see if our build process handles this now.

ohadlevy commented:
updated, removed datatables.net-bs and node-sass from the pr.

On Tue, Jun 19, 2018 at 2:29 PM Ewoud Kohl van Wijngaarden <
notifications@github.com> wrote:

> *@ekohl* commented on this pull request.
>
> I've opened theforeman/foreman-packaging#2673
> <https://github.com/theforeman/foreman-packaging/pull/2673>. This is the
> first package we handle where something depends on @x/y where everything
> else just had x as names. Let's see if our build process handles this now.
> ------------------------------
>
> In package.json
> <https://github.com/theforeman/foreman/pull/5708#discussion_r196385844>:
>
> > @@ -61,7 +61,7 @@
>      "babel-polyfill": "^6.26.0",
>      "brace": "^0.10.0",
>      "datatables.net": "~1.10.12",
> -    "datatables.net-bs": "~1.10.12",
> +    "datatables.net-bs": "^2.1.1",
>
> This specific version is marked as deprecated on NPM. Are you sure this is
> the correct version? https://www.npmjs.com/package/datatables.net-bs
> lists 1.10.16 as the current version.
> ------------------------------
>
> In package.json
> <https://github.com/theforeman/foreman/pull/5708#discussion_r196386103>:
>
> > @@ -39,16 +39,16 @@
>      "jest-cli": "^20.0.0",
>      "jsdom": "^9.5.0",
>      "lodash-webpack-plugin": "^0.11.4",
> -    "node-sass": "^4.5.0",
> +    "node-sass": "^4.9.0",
>
> This is a rather complicated package to build. Could we submit this as a
> separate PR so we can get all the other changes in first?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/theforeman/foreman/pull/5708#pullrequestreview-129925839>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABOx96hIN8zv8uoyCvUuRT2vqwh_bUrks5t-OCmgaJpZM4Uq0Mf>
> .
>

ohadlevy commented:
also, would you like a redmine issue (one or more?) for these?

ekohl reviewed (approved):
Personally I don't think a redmine issue is that important in this case but it could be considered good practice to always do so.

mmoll commented:
Test failures are unrelated.

I didn't actually test this on a foreman instance and leave it to the RPM people to actually merge.

ohadlevy commented:
@ekohl bump?

ekohl commented:
@ohadlevy I already gave an approval
